static/js/directives/time-ago-auto-update.js

```diff
@@ -0,0 +1,59 @@
+var moment = require('moment');
+
+/**
+ * Directive for an auto-updating relative time ago date.
+ * Renders only text and auto-updates it. For better display should be
+ * used in other directives.
+ */
+angular.module('inboxDirectives').directive('timeAgoAutoUpdate', ['$interval', 'FormatDate', function($interval, FormatDate) {
+  'use strict';
+  'ngInject';
+
+  function getRelativeDateString(date, options) {
+    if (options.age) {
+      return FormatDate.age(date);
+    }
+    if (!options.withoutTime && date.isSame(moment(), 'day')) {
+      return FormatDate.time(date);
+    }
+    return FormatDate.relative(date, options);
+  }
+
+  function renderTime(element, date, options) {
+    element.text(getRelativeDateString(date, options));
+  }
+
+  function link(scope, element) {
+    var timeoutId;
+    var momentDate = moment(scope.date);
+
+    var options = {};
+    options.age = scope.age;
+    options.withoutTime = scope.withoutTime;
+
+    element.on('$destroy', function() {
+      $interval.cancel(timeoutId);
+    });
+
+    // Render it immediately.
+    renderTime(element, momentDate, options);
+
+    // start the UI update process; save the timeoutId for canceling
+    timeoutId = $interval(function() {
+      renderTime(element, momentDate, options);
+    }, 1000);
```

Comment:
I'm concerned about performance here as we could have 100s of these firing every second and we run on very low end devices. Raising it to once per minute would be a better compromise.

Alternatively the interval could increase as the time gets further away, for example, if the date is two weeks away checking once a day would be fine. If it's variable it may need $timeout instead of $interval so it can be updated each time.

Is there a way of batching this together so we don't get 100s of digest cycles per second? Or will angular look after that for us under the hood?

Reply:
Yeah, that's a valid concern. I've moved that to a single entity, which handles this and future global timers.